node/internal.ts

```diff
@@ -182,13 +182,12 @@ export function _loc(key: string, ...param: any[]): string {
     if (!_libResourceFileLoaded) {
         // merge loc strings from azure-pipelines-task-lib.
         var libResourceFile = path.join(__dirname, 'lib.json');
+        _libResourceFileLoaded = true;
```

Comment:
Setting to true before doing anything?

Reply:
Can we add tests for this?

Reply:
> Setting to true before doing anything?

Yeah, right now if [_loc](https://github.com/Microsoft/azure-pipelines-task-lib/blob/b9602e028255065aed92c64fa5c07f235542b1bc/node/internal.ts#L181) is called for the first time, it calls [_loadLocStrings](https://github.com/Microsoft/azure-pipelines-task-lib/blob/b9602e028255065aed92c64fa5c07f235542b1bc/node/internal.ts#L185) before setting it to true. This normally makes sense, but if [_exist(resourceFile)](https://github.com/Microsoft/azure-pipelines-task-lib/blob/b9602e028255065aed92c64fa5c07f235542b1bc/node/internal.ts#L102) returns false, we [call _loc](https://github.com/Microsoft/azure-pipelines-task-lib/blob/b9602e028255065aed92c64fa5c07f235542b1bc/node/internal.ts#L135) again which kicks off the loop. By setting it to true we avoid this loop, and this is the only place its checked so changing it early shouldn't have any side effects. Basically we're saying that since we've started trying to load libResourceFile we're not going to try to do it again.

As I think about it though, it probably makes more sense to just not call _loc [here](https://github.com/Microsoft/azure-pipelines-task-lib/blob/b9602e028255065aed92c64fa5c07f235542b1bc/node/internal.ts#L135) since its guaranteed not to be able to find the correct localization anyways. I'll update accordingly.